keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -266,8 +266,16 @@ class SqlArtifactRepository(
           .and(ENVIRONMENT_ARTIFACT_VERSIONS.APPROVED_AT.isNotNull)
           .and(ENVIRONMENT_ARTIFACT_VERSIONS.PROMOTION_STATUS.ne(VETOED.name))
           .orderBy(ENVIRONMENT_ARTIFACT_VERSIONS.APPROVED_AT.desc())
-          .limit(1)
-          .fetchOne(ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_VERSION)
+          .limit(20)
+          .fetch(ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_VERSION)
+          .let {
+            when (it.isNotEmpty()) {
+              true -> it
+                .sortedWith(artifact.versioningStrategy.comparator)
```

Comment:
This changes `latestVersionApprovedIn` from returning the most recently approved version (which may not be the newest version, approval isn't necessarily linear!) to the most recent version within the set of the most recently approved 20 versions, according to the artifacts versioning strategy.